PythonVisualizations/AdvancedSorting.py

```diff
@@ -0,0 +1,570 @@
+                kIndex = self.createIndex(i, 'k')
+                callEnviron |= set(kIndex)
+            
+                # Slide values from right to left to fill gap
+                for j in range(i+1, len(self.list)):
+                    self.assignElement(j, j - 1, callEnviron)
+                    self.fixCells() # retain the correct shape
+                    self.moveItemsBy(kIndex, (self.CELL_WIDTH, 0), sleepTime=0.01)
+                    self.wait(0.1)
+
+                # remove the last item in the list
+                n = self.list.pop()
+                # delete the associated display objects
+                self.canvas.delete(n.display_shape)
+                self.canvas.delete(n.display_val)
+                
+                # update window
+                self.wait(0.3)
+
+                self.cleanUp(callEnviron)
+                return True
+
+            # if not found, then move the index over one cell
+            self.moveItemsBy(indexDisplay, (self.CELL_WIDTH, 0), sleepTime=0.01)
+            if self.wait(0.1):
+                break
+
+        # Animation stops
+        self.cleanUp(callEnviron)
+        return None
+
+    
+    def new(self, val):
+        # if changeSize is equal to False
+        if self.changeSize:
+            # set CELL_WIDTH to be equal to CELL_SIZE
+            self.CELL_WIDTH = self.CELL_SIZE
+            # keep changeSize equal to False
+            self.changeSize = False
+        # if there is no room on the canvas  
+        if self.window.winfo_width() <= self.ARRAY_X0 + (val+1) * self.CELL_WIDTH:
+            # change the cell width
+            self.CELL_WIDTH = 20
+            # changeSize is set to True 
+            self.changeSize = True
+            # even when the cell width is smaller, return False if there is no room on the canvas 
+            if self.window.winfo_width() <= self.ARRAY_X0 + (val+1) * self.CELL_WIDTH:
+                return False
+            
+        callEnviron = self.createCallEnvironment()                
+        # Clear Array and reset size and list
+        self.canvas.delete("all")
+        self.size = val
+        self.list = []        
+        
+        for i in range(val):  # Draw new grid of cells
+            self.createArrayCell(i) 
+        
+        self.window.update()
+        self.cleanUp(callEnviron)
+        # return True upon success 
+        return True
+        
+    def randomFill(self):
+        callEnviron = self.createCallEnvironment()        
+        # Clear the list so new values can be entered
+        self.list=[] 
+        size = self.size
+        
+        # Create a list of random numbers and sort them
+        a = [random.randrange(90) for i in range(size)]
+        
+        # Append and draw them to the list and draw them
+        for i in a:
+            self.list.append(drawable(i))
+        
+        self.display()         
+        self.cleanUp(callEnviron)      

```

Comment:
Let's move the Random Fill operation to the `SortingBase` class to share it with other visualizations.  I think it can also be simplified to:
```
        self.list=[drawable(random.randrange(90)) for in range(self.size)]
```

Reply:
Fixed in commit 0c51fde